packages/react-i18n/docs/react_i18n_schema.md

```diff
@@ -0,0 +1,284 @@
+# Introduction <!-- omit in toc -->
+
+React I18n is a schema comprised of a structure of nested key-value pairs (KVPs). This document captures the nuances of the constraints placed on the schema by the different levels of concern.
+
+<!-- Created using "Markdown All in One" extension for VS Code -->
+- [History](#history)
+  - [Version 1.0](#version-10)
+- [Versioning](#versioning)
+- [Basic Structure](#basic-structure)
+- [Pluralization](#pluralization)
+  - [Reserved pluralization keys](#reserved-pluralization-keys)
+  - [Invalid pluralization keys](#invalid-pluralization-keys)
+  - [Detecting pluralization contexts](#detecting-pluralization-contexts)
+- [Serialization](#serialization)
+  - [Serialization to JSON](#serialization-to-json)
+    - [Comments](#comments)
+  - [Serialization to JSON+Comments](#serialization-to-jsoncomments)
+- [Comparison with Rails I18n](#comparison-with-rails-i18n)
+    - [Root locale key](#root-locale-key)
+    - [Pluralization](#pluralization-1)
+    - [Interpolation](#interpolation)
+
+# History
+
+## Version 1.0
+
+React I18n's schema is loosely based on the [Rails I18n](https://guides.rubyonrails.org/i18n.html) schema.
+
+Prior to the creation of this document, the React I18n schema was not documented.
+With the creation of this document, the schema was specified and given the version identifier `1.0`.
+
+# Versioning
+
+The version identifier for the schema is made of two parts, `<major>` and `<minor>`, separated by a `.`: `<major>.<minor>`.
+
+the `<major>` version number will be incremented any time a change is introduced that would mean that existing parsers would break while parsing the new schema. Put another way, software that supports parsing version `x.y` of the schema will always be able to parse any version `x.z`, where `z >= y`.
```

Comment:
> Put another way, software that supports parsing version `x.y` of the schema will always be able to parse any version `x.z`, where `z >= y`.

True, but should these two paragraphs be re-organized so that this is at the end? This example also demonstrates the meaning of a `<minor>` version change.

Reply:
🤷 ... I'm going to keep it this way.